src/handlers/listHandler.ts

```
@@ -0,0 +1,135 @@
+import fs from "fs";
+import path from "path";
+import { Item, ShoppingList } from "../types/types";
+
+class listHandler {
+  constructor() {}
+
+  getAll = async (): Promise<ShoppingList[]> => {
+    try {
+      const content = await fs.promises.readFile(
+        path.resolve(__dirname, "../data/lists.json"),
+        "utf-8"
+      );
+      const lists = JSON.parse(content || "[]");
+      return lists;
+    } catch (err) {
+      console.log(err.message);
+      return [];
```

here you should return an error message instead of empty array

---

user won't know that sth failed if not